roofit/batchcompute/inc/RunContext.h

```diff
@@ -39,19 +39,23 @@ struct RunContext {
   /// Move a RunContext. All spans pointing to data retrieved from the original remain valid.
   RunContext(RunContext&&) = default;
   RooSpan<const double> getBatch(const RooArgProxy& proxy) const;
-  RooSpan<const double> getBatch(const RooAbsReal* owner) const;
+  RooSpan<const double> getBatch(const RooAbsArg* owner) const;
   /// Retrieve a batch of data corresponding to the element passed as `owner`.
-  RooSpan<const double> operator[](const RooAbsReal* owner) const { return getBatch(owner); }
-  RooSpan<double> getWritableBatch(const RooAbsReal* owner);
-  RooSpan<double> makeBatch(const RooAbsReal* owner, std::size_t size);
+  RooSpan<const double> operator[](const RooAbsArg* owner) const { return getBatch(owner); }
+  RooSpan<double> getWritableBatch(const RooAbsArg* owner);
+  RooSpan<double> makeBatch(const RooAbsArg* owner, std::size_t size);
 
   /// Clear all computation results without freeing memory.
-  void clear() { spans.clear(); rangeName = nullptr; }
+  void clear();
 
   /// Once an object has computed its value(s), the span pointing to the results is registered here.
-  std::unordered_map<const RooAbsReal*, RooSpan<const double>> spans;
+  std::map<const RooAbsArg*, RooSpan<const double>> spans;
+  std::map<const RooAbsArg*, const double*> spansCuda;
+
   /// Memory owned by this struct. It is associated to nodes in the computation graph using their pointers.
-  std::unordered_map<const RooAbsReal*, std::vector<double>> ownedMemory;
+  std::map<const RooAbsArg*, std::vector<double>> ownedMemory;
+  std::map<const RooAbsArg*, double*> ownedMemoryCuda;
```

Comment:
No, this doesn't work, because the memory has to be freed with `cudaFree` here. In the future I want to make all of this more automatic with some wrapper types for this memory management, but so far it's easier to stay like this.